letsencrypt/DOCS.md

```diff
@@ -302,6 +302,51 @@ dns:
 ```
 Use `ovh_endpoint: ovh-ca` for north america region.
 
+### RFC2136
+You will need to set up a server with RFC2136 (Dynamic Update) support with a TKEY (to authenticate the updates).  How to do this will vary depending on the DNS server software in use.  For Bind9, you first need to first generate an authenticate key by running
+```
+# dnssec-keygen -a HMAC-SHA512 -b 512 -n HOST letsencrypt
+Kletsencrypt.+165+20675
+```
+The key file (Kletsencrypt.+165+20675.key in this example) looks like the following:
+```
+# cat Kletsencrypt.+165+20675.key
+letsencrypt. IN KEY 512 3 165 Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==
+```
+You don't need to publish this; just copy the key data into your named.conf file:
+```
+key "letsencrypt" {
+  algorithm hmac-sha512;
+  secret "Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==";
+};
+```
+And ensure you have an update policy in place in the zone that uses this key to enable update of the correct domain
+```
+   update-policy {
+      grant letsencrypt name _acme-challenge.hassio.dom.ain. txt;
+   };
+```
+
+For this add-on you will need to supply all the rfc2136_* options.  Note that the rfc2136_port item is required (there is no default port in the add-on) and, most importantly, the the port number must be quoted because the add-on is expecting it to be a string and not a number.  Also be sure to copy in the key so certbot can authenticate to the DNS server.
+
+An example configuration
+```yaml
+email: your-mail@dom.ain
+domains:
+  - hassio.dom.ain
+certfile: fullchain.pem
+keyfile: privkey.pem
+challenge: dns
+dns:
+  provider: dns-rfc2136
+  rfc2136_server: dns-server.dom.ain
```

Comment:
I couldn't get this add-on to work when I used a FQDN. I had to change the line from:
`rfc2136_server: dns-server.dom.ain`
to:
`rfc2136_server: 123.123.123.123`
Before it would work.

Reply:
That's odd, because it works fine for me using FQDN.
My issue was that it did NOT work if I ignored the (default port 53) setting.

Reply:
hmm, I went back and tried changing the rfc2136_server from an IP to my DNS server's FQDN and the Let's Encrypt add-on failed again producing a number of exceptions. I'm certainly no expert here so I may have something else incorrectly configured in my case. It is working great with the IP though!

Reply:
What exceptions did it give you? 

I am happy to update this PR to suggest that one may need to supply the IP address in lieu of the FQDN.

Reply:
Sure, my log from the Let's Encrypt add-on is here: https://pastebin.com/7r7JnV0V
For some reason the add-on doesn't show the start of the log so I don't see the initial exception. What I put on pastebin is everything I could see.

Reply:
@EmnaX -- I'm sorry, but I think we need more of the log.  The initial failure is higher up, before what you posted.  :(
Are you willing to share the FQDN of your DNS server and how you had it in your config when it failed?